.github/workflows/pr-review-check.yml

```
@@ -0,0 +1,51 @@
+name: "PR Review check"
+
+on:
+  workflow_call:
+    inputs:
+      trigger-labels:
+        description: JSON object mapping the list of labels that should trigger the validation
+        required: true
+        type: string
+      milestone-name:
+        description: Milestone to set on the PR when validation is triggered
+        required: true
+        type: string
+    secrets:
+      BOT_GITHUB_TOKEN:
+        description: >
+          Token used to enable auto-merge, comment on the PR, and change PR milestone/label.
+          This token CANNOT be the default `GITHUB_TOKEN` if auto-merge is set, otherwise the merge of the PR will not trigger a build.
+          Or similarly, if labeled and milestoned events are expected to trigger a new build.
+        required: true
+
+env:
+  DEPENDABOT_CREATOR: 'dependabot[bot]'
+
+permissions:
+  contents: read          # Required to read repository content
+  pull-requests: write    # Required to edit PR (milestones, labels, auto-merge)
+  issues: write          # Required to create comments on PRs
+
+jobs:
+  check:
+    if: github.secret_source != 'None'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Trigger validation on approval
+        if: github.event_name == 'pull_request_review'
+        uses: Alfresco/alfresco-build-tools/.github/actions/github-trigger-approved-pr@c295ecd6ca6b9c473dcada1e0478cdeaba79d5ea # v9.0.0
+        with:
+          creator: ${{ env.DEPENDABOT_CREATOR }}
+          github-token: ${{ secrets.BOT_GITHUB_TOKEN }}
+          milestone-on-approval: ${{ inputs.milestone-name }}
+          auto-merge-on-approval: true
+
+      - name: Trigger validation on labeling, unless token is readonly (dependabot/fork) or milestone
```

```suggestion
      - name: Trigger validation on labeling when token is not readonly and milestone is set
```

@atchertchian from what I understand this would be a better name for the step? Unless you expected it to work the way you typed it in the natural language step description, and the condition just below is wrong instead (`&& inputs.milestone-name != ''` instead of `|| ...`?)